File: 0520-detect-capital/0520-detect-capital.cpp
class Solution {
public:
    bool detectCapitalUse(string word) {
        int c=0;
        for(int i=word.size()-1;i>=0;i--){
            if(isupper(word[i])) c++;
            
        }
        if(c==word.size()) return true;
        else if(c==0 && word.size()>=1) return true;
        else if(c==1 && isupper(word[0])) return true;
        return false;
    }
};

File: 0520-detect-capital/0520-detect-capital.java
class Solution {
    public boolean detectCapitalUse(String word) {
        int c=0;
        for(int i=word.length()-1;i>=0;i--){
            if(word.charAt(i)==Character.toUpperCase(word.charAt(i))) c++;
            
        }
        if(c==word.length()) return true;
        else if(c==0 && word.length()>=1) return true;
        else if(c==1 && word.charAt(0)==Character.toUpperCase(word.charAt(0))) return true;
        return false;
    }
}